### ISSUE TITLE
Add info to right-hand sidebar for "Add a new sentence" covering unsupported language

### ISSUE THREAD

**Issue by alanfgh** [Tatoeba/tatoeba2#549]
On the Wall, Silja [http://tatoeba.org/eng/wall/show_message/21352#message_21352](pointed out) that we don't anticipate the case where the user wants to add a sentence in a language that's not supported. I think we should add either information to the right-hand sidebar about this, or a link to such a discussion on the wiki, or both.

**Comment by jiru** [comment]
This makes me wonder if we couldn’t change the string “unknown language” into something more appropriate like “unsupported language” or “other language”. Correct me if I’m wrong, but users basically know the language they are adding sentence into, so it’s not unknown, at least to someone, the author, who may indicate the actual language by other means like comments or lists. When the string is displayed in other parts of the website, like flag mouseovers, I think the same reasoning makes sense to a certain extend.

**Comment by alanfgh** [comment]
Just to clarify to others who may be reading this: The drop-down list for the "Add new sentence" page contains only "Auto detect" and the names of the languages that are currently supported in the corpus. The drop-down list for "Show sentences in:" on the "Browse language by" page contains an item "unknown language" plus the names of the languages that are currently supported in the corpus. To address the issues that we've been discussing, we should do the following:

(1) For the "Add new sentence" page, in addition to "Auto detect" and the supported languages, we should also display "other language".

(2) For the "Show sentences in:" drop-down list on the "Browse language by" page, we should change the "unknown language" string to "other language".

The fix for (1) would involve:
(a) changing translationsArray in views/helpers/languages.php to contain an additional item with the key "other" and the label "other language"
(b) adding code to save_translation() in app/controllers/sentences_controller.php to handle the case where $translationLang == 'other'
(c) possibly adding code to wrapper_save_sentence() in controllers/components/common_sentence.php to handle the case where $translationLang == 'other'

The fix for (2) would involve changing the string "unknown language" in unknownLanguagesArray in views/helpers/languages.php to "other language". (It would also be a good idea to change the name of the unknownLanguagesArray to something else at the same time.)

We should also extend the information in the right-hand sidebar.

**Comment by alanfgh** [comment]
> This makes me wonder if we couldn’t change the string “unknown language” into something more appropriate like “unsupported language” or “other language”.

Yes. Of the two, I much prefer "other language".

**Comment by alanfgh** [comment]
@trang, @jiru, @loolmeh, I'd like some feedback on some proposed solutions.

My goal is to get people who want to add a sentence in a language that we don't support (yet) to:
(a) read the right sidebar (where I'm going to put or link to information about adding a sentence in an unsupported language)
(b) think twice about whether they really want to add that sentence

So there's a question about what the drop-down list should say. Here's one possibility:

Auto detect
Other language
Abkhaz
Afrikaans
...

And here's another, which is less streamlined and more geeky but does emphasize that someone should read the sidebar before adding a sentence in an unsupported language:

Auto detect
Other language (see sidebar)
Abkhaz
Afrikaans

It would be possible to put "Other language" at the end of the list, but I'd rather not, since I suspect no one will ever see it.

I'd rather stay away from "Unsupported language" because it sounds too final, like something that we'll never support. But you may disagree. Also, maybe there's some other string that would work. 

We could possibly put "Auto detect" and/or "Other language" in italics (though I don't know how to do that at the moment).

Any feedback would be welcome.

**Comment by jiru** [comment]
I agree about all your renaming and drop down modification, but not the right-hand sidebar.

To be honest, I hate most of these right-hand sidebars, I think they are bad by nature in the first place, because [people don’t read them and they shouldn’t have to](http://www.joelonsoftware.com/uibook/chapters/fog0000000062.html). Worse, most of the people who are going to read it will likely loose their time because they are very unlikely to want to add a sentence in a not yet supported language.

How about adding a simple link like “Your language is not in the list?” next to the drop down, that would link to or show the relevant information.

**Comment by alanfgh** [comment]
I see your point, but there are right-hand sidebars throughout the UI and I don't want to break that pattern unless we change it everywhere. In any case, I have found them useful on occasion.

**Comment by trang** [comment]
I don't think it would break the pattern to add a link "Your language is not in the list?".
I agree with gillux that it's more intuitive to have such a link.

Just some quick examples:
http://prntscr.com/5rxxbq
http://prntscr.com/5rxxje

**Comment by alanfgh** [comment]
It occurs to me that "Unlisted language" would be an alternative to "Other language". Arguably, it's preferable, since "other" is often used in a context where one can specify a string that is not already found in a list, which is not the case here. Also, "Unlisted" would tie into the wording of the question "Your language is not in the list?" 

Note that regardless of whether we use "Unlisted" or "Other", we probably should make "language" plural in the "Show sentences in" case (that is, "Unlisted languages" or "Other languages").
